mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/master/resourcecluster/ResourceClusterScalerActor.java

```diff
@@ -138,32 +141,30 @@ public Receive createReceive() {
             ReceiveBuilder
                 .create()
                 .match(TriggerClusterUsageRequest.class, this::onTriggerClusterUsageRequest)
+                .match(TriggerClusterRuleRefreshRequest.class, this::onTriggerClusterRuleRefreshRequest)
+                .match(GetRuleSetRequest.class,
+                    req -> getSender().tell(
+                        GetRuleSetResponse.builder().rules(ImmutableMap.copyOf(this.skuToRuleMap)).build(), self()))
                 .match(GetClusterUsageResponse.class, this::onGetClusterUsageResponse)
                 .match(GetClusterIdleInstancesResponse.class, this::onGetClusterIdleInstancesResponse)
+                .match(GetRuleSetResponse.class,
+                    s -> log.info("[{}] Refreshed rule size: {}", s.getClusterID(), s.getRules().size()))
                 .match(Ack.class, ack -> log.info("Received ack from {}", sender()))
                 .build();
     }
 
     private void init() {
-        this.storageProvider.getResourceClusterScaleRules(this.clusterId.toString())
-                .thenAccept(rules ->
-                     rules
-                        .getScaleRules().values()
-                        .forEach(rule -> {
-                            log.info("Adding scaleRule: {}", rule);
-                            this.skuToRuleMap.put(
-                                rule.getSkuId(),
-                                new ClusterAvailabilityRule(rule, this.clock));
-                        }));
-
-        this.storageProvider.getResourceClusterSpecWritable(this.clusterId.toString())
-            .thenAccept(specWritable ->
-                this.skuMapperRef.set(new MachineDefinitionToSkuMapper(specWritable.getClusterSpec())));
+        this.fetchRuleSet();
 
         getTimers().startTimerWithFixedDelay(
             "ClusterScaler-" + this.clusterId,
             new TriggerClusterUsageRequest(this.clusterId),
             scalerPullThreshold);
+
+        getTimers().startTimerWithFixedDelay(
+            "ClusterScalerRuleFetcher-" + this.clusterId,
+            new TriggerClusterRuleRefreshRequest(this.clusterId),
```

Comment:
Wondering why the rules are being refreshed periodically? Isn't the trigger refresh message sent to the actor good enough?

Reply:
I added this before integrating with API update routes and decide to keep it in case any failure happen in the other route. The freq is low enough and shouldn't be any burden to the process.